controllers/azurecluster_reconciler.go

```diff
@@ -131,9 +136,51 @@ func (s *azureClusterService) Reconcile(ctx context.Context) error {
 		return errors.Wrap(err, "failed to reconcile bastion")
 	}
 
+	if err := s.reconcileTags(ctx); err != nil {
+		return errors.Wrap(err, "failed to reconcile resource tags")
+	}
+
+	return nil
+}
+
+func (s *azureClusterService) reconcileTags(ctx context.Context) error {
+	// check that the resource group is not BYO.
+	managed, err := s.isResourceGroupManaged(ctx)
+	if err != nil {
+		return errors.Wrap(err, "could not get resource group management state")
+	}
+	if !managed {
+		s.scope.V(2).Info("Should not update resource group tags in unmanaged mode")
+		return azure.ErrNotOwned
+	}
+
+	if err := s.tagsSvc.Reconcile(ctx); err != nil {
+		return errors.Wrap(err, "unable to update tags")
+	}
+
 	return nil
 }
 
+func (s *azureClusterService) isResourceGroupManaged(ctx context.Context) (bool, error) {
```

Comment:
@CecileRobertMichon this is one of the ideas that I thought about, but yeah, does not look very great, especially if we start doing this check for each resource we add to tag spec it will get bloated up, also we still have to work out how to skip some reconciliations when we have more than just resource group and when they are not managed and when we don't want to update resource's tags if they are not managed